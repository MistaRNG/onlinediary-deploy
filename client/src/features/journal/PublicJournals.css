.back-to-editor-button {
    margin-bottom: 20px;
}

.dark .back-to-editor-button {
    background-color: #444;
    color: #e5e5ea;
}

.dark .back-to-editor-button:hover {
    background-color: #333;
}

.modal-container .bp4-dialog {
    border-radius: 8px;
    z-index: 1100 !important;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    background-color: var(--body-background-color);
    font-size: 1.2em !important;
}

.dark .modal-container .bp4-dialog {
    background-color: #1c1c1e !important;
    color: #e5e5ea !important;
}

.page-container {
    display: flex;
    flex-direction: column;
    height: 100vh;
}

.header-container {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    background-color: var(--body-background-color);
    z-index: 500;
    padding: 1em;
}

.content-container {
    flex-grow: 1;
    overflow-y: auto;
    padding: 1em;
    margin-top: 205px !important;
}

.public-journals-container {
    display: flex;
    flex-wrap: wrap;
    gap: 1em;
    justify-content: flex-start;
    align-content: flex-start;
}

.public-journal-card {
    flex: 1 1 calc(33.333% - 1em);
    max-width: calc(33.333% - 1em);
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: auto;
    padding: 15px;
    margin-bottom: 1em;
    min-width: 250px;
    transition: transform 0.2s;
}

.public-journal-card:hover {
    transform: scale(1.03);
}

.public-journal-card h5 {
    margin-bottom: 0.5em;
    font-size: 1.1em;
}

.public-journal-card p {
    margin-bottom: 1em;
    font-size: 0.9em;
}

.search-input {
    margin-bottom: 20px;
    width: 100%;
    max-width: 400px;
}

.card-actions {
    display: flex;
    justify-content: space-between;
  }

.bp4-dialog-footer {
display: flex;
justify-content: space-between;
}  

.like-button {
    padding: 8px 16px;
    border-radius: 4px;
    cursor: pointer;
    transition: all 0.3s ease;
}

.like-button.dark-mode {
    background-color: #383e47;
    color: #ffffff;
    border: 2px solid #394b59;
}

.like-button.dark-mode.liked {
    background-color: #272b2f;
    color: #f5f8fa;
    border-color: #30404d;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
    border: 3px solid #394b59;
}

.like-button.dark-mode:hover {
    background-color: #2c313a;
}

.like-button.dark-mode.liked:hover {
    box-shadow: 0 0 12px rgba(0, 0, 0, 0.7);
}

.like-button.light-mode {
    background-color: #f0f0f0;
    color: #000000;
    border: 1px solid #c0c0c0;
}

.like-button.light-mode.liked {
    background-color: #d8d8d8;
    color: #000000;
    border-color: #838383;
    box-shadow: 0 0 10px rgba(144, 144, 144, 0.5);
    border: 3px solid #c0c0c0;
}

.like-button.light-mode:hover {
    background-color: #e0e0e0;
}

.like-button.light-mode.liked:hover {
    box-shadow: 0 0 12px rgba(180, 180, 180, 0.7);
}

.comments-container {
    max-height: 200px;
    overflow-y: auto;
    padding: 10px;
    border-radius: 8px;
    background-color: var(--body-background-color);
    border: 1px solid var(--border-color);
}


@media (max-width: 768px) {
    .public-journal-card {
        flex: 1 1 calc(50% - 1em);
        max-width: calc(50% - 1em);
    }
}

@media (max-width: 480px) {
    .public-journal-card {
        flex: 1 1 calc(100% - 1em);
        max-width: calc(100% - 1em);
    }
}

@media (min-width: 1024px) {
    .public-journal-card {
        flex: 1 1 calc(25% - 1em);
        max-width: calc(25% - 1em);
    }
}

@media (max-height: 850px) {
    .modal-container {
        max-height: 80vh !important;
    }
}

.bp4-overlay * {
    outline: none !important;
}
