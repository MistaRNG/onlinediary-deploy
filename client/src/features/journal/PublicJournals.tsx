import "./PublicJournals.css";
import React, { useEffect, useState } from "react";
import { Card, Elevation, Button, Dialog } from "@blueprintjs/core";
import { getPublicJournals } from "features/journal/journalSlice";
import { useDispatch, useSelector } from "react-redux";
import { AppDispatch, RootState } from "app/store";
import { EditorState, convertFromRaw, Editor } from "draft-js";
import axios from "axios";
import moment from "moment";
import useMode from "common/hooks/useMode";
import { useNavigate } from "react-router-dom";
import SearchInput from "./components/Search/Input";
import LogoutButton from "features/auth/LogoutButton";
import useCurrentUser from "common/hooks/useCurrentUser";

const PublicJournals: React.FC = () => {
  const { darkMode } = useMode();
  const dispatch: AppDispatch = useDispatch();
  const journals = useSelector((state: RootState) => state.journals.data);
  const navigate = useNavigate();
  const { username } = useCurrentUser();
  const [isPasswordInputVisible, setIsPasswordInputVisible] = useState(false);

  useEffect(() => {
    const checkPasswordInput = () => {
      const passwordInputVisible = !!document.querySelector('input[type="password"]');
      setIsPasswordInputVisible(passwordInputVisible);
    };

    checkPasswordInput();

    const observer = new MutationObserver(checkPasswordInput);
    observer.observe(document.body, { childList: true, subtree: true });

    return () => observer.disconnect();
  }, []);

  useEffect(() => {
    const fetchPublicJournals = async () => {
      await dispatch(getPublicJournals());
    };

    fetchPublicJournals();
  }, [dispatch]);

  const [searchQuery, setSearchQuery] = useState("");
  const [likes, setLikes] = useState<Record<number, number>>({});
  const [userLiked, setUserLiked] = useState<Record<number, boolean>>({});
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [selectedJournal, setSelectedJournal] = useState<any>(null);
  const [editorState, setEditorState] = useState<EditorState | null>(null);
  const [comments, setComments] = useState<any[]>([]);
  const [newComment, setNewComment] = useState<string>("");

  const fetchLikes = async (journalId: number) => {
    try {
      const response = await axios.get(`http://localhost:3003/api/likes/${journalId}`, {
        withCredentials: true,
      });
      setLikes((prev) => ({ ...prev, [journalId]: response.data.count || 0 }));
      setUserLiked((prev) => ({ ...prev, [journalId]: response.data.userLiked || false }));
    } catch (error) {
      if (error.response && error.response.status === 404) {
        setLikes((prev) => ({ ...prev, [journalId]: 0 }));
        setUserLiked((prev) => ({ ...prev, [journalId]: false }));
      } else {
        console.error("Error fetching likes:", error);
      }
    }
  };

  const fetchComments = async (journalId: number) => {
    try {
      const response = await axios.get(`http://localhost:3002/api/comments/${journalId}`);
      setComments(response.data);
    } catch (error) {
      console.error("Error fetching comments:", error);
    }
  };

  const handleLike = async (journalId: number) => {
    try {
      if (!userLiked[journalId]) {
        await axios.post(
          "http://localhost:3003/api/likes",
          { journal_id: journalId },
          { withCredentials: true }
        );
  
        setUserLiked((prev) => ({ ...prev, [journalId]: true }));
        setLikes((prev) => ({ ...prev, [journalId]: prev[journalId] + 1 }));
      } else {
        await axios.delete("http://localhost:3003/api/likes", {
          data: { journal_id: journalId },
          withCredentials: true,
        });
  
        setUserLiked((prev) => ({ ...prev, [journalId]: false }));
        setLikes((prev) => ({ ...prev, [journalId]: prev[journalId] - 1 }));
      }
    } catch (error) {
      console.error("Error liking/unliking:", error);
    }
  };

  const openJournalModal = (journal: any) => {
    const formattedDate = moment(journal.date).format("dddd, MMMM D, YYYY");

    setSelectedJournal({
      ...journal,
      formattedDate,
    });

    const contentState = convertFromRaw(journal.content);
    const editorState = EditorState.createWithContent(contentState);
    setEditorState(editorState);

    setIsModalOpen(true);

    fetchComments(journal.id);
  };

  const closeJournalModal = () => {
    setIsModalOpen(false);
    setSelectedJournal(null);
    setComments([]);
    setNewComment("");
  };

  const handleCommentSubmit = async () => {
    if (newComment.trim() === "") return;
  
    try {
      const response = await axios.post(
        "http://localhost:3002/api/comments",
        {
          journal_id: selectedJournal.id,
          content: newComment,
        },
        {
          withCredentials: true,
        }
      );
  
      setComments([...comments, { ...response.data, edited: false }]);
      setNewComment("");
    } catch (error) {
      console.error("Error posting comment:", error);
    }
  };

  const handleEditComment = async (commentId: number, currentContent: string) => {
    const newContent = prompt("Edit your comment:", currentContent);
    if (newContent && newContent.trim() !== "") {
      try {
        await axios.put(`http://localhost:3002/api/comments/${commentId}`, {
          content: newContent,
        });
        fetchComments(selectedJournal.id);
      } catch (error) {
        console.error("Error updating comment:", error);
      }
    }
  };

  const handleDeleteComment = async (commentId: number) => {
    if (window.confirm("Are you sure you want to delete this comment?")) {
      try {
        await axios.delete(`http://localhost:3002/api/comments/${commentId}`);
        fetchComments(selectedJournal.id);
      } catch (error) {
        console.error("Error deleting comment:", error);
      }
    }
  };

  useEffect(() => {
    Object.values(journals)
      .filter((journal: any) => journal.id)
      .forEach((journal: any) => {
        fetchLikes(journal.id);
      });
  }, [journals]);  

  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchQuery(event.target.value);
  };

  const cancelSearch = () => {
    setSearchQuery("");
  };

  const filteredJournals = Object.values(journals).filter((journal: any) =>
    journal.title?.toLowerCase().includes(searchQuery.toLowerCase()) ||
    moment(journal.date).format("dddd, MMMM D, YYYY").toLowerCase().includes(searchQuery.toLowerCase())
  );

  return (
    <div className={darkMode ? "dark" : "bright"}>
      <LogoutButton />
      <div className="page-container">
        {!isPasswordInputVisible && (
          <>
            <div
              className="header-container"
              style={{
                position: "fixed",
                filter: isModalOpen ? "brightness(0.5)" : "none",
                margin: "70px 0 0 1px",
                zIndex: 1,
                display: "block",
              }}
            >
              <Button
                icon="arrow-left"
                text="Back to Editor"
                onClick={() => navigate(`/journal/${moment().format("YYYY-MM-DD")}`)}
                className="back-to-editor-button"
                style={{
                  marginBottom: "30px",
                  display: "block",
                }}
              />
              <SearchInput
                search={searchQuery}
                changeHandler={handleSearchChange}
                cancel={cancelSearch}
              />
            </div>
            <h2
              style={{
                textAlign: "center",
                width: "100%",
                marginTop: "20px",
                position: "fixed",
                top: "30px",
                left: "50%",
                transform: "translateX(-50%)",
                zIndex: 1,
              }}
            >
              Public entries
            </h2>
          </>
        )}
        <div
          className="content-container"
          style={{
            filter: isModalOpen ? "brightness(0.5)" : "none",
            marginTop: "100px",
            zIndex: 0,
            position: "relative",
          }}
        >
          <div className="public-journals-container">
            {filteredJournals.length ? (
              filteredJournals.map((journal: any) => {
                return (
                  <Card
                    key={journal.id}
                    interactive={true}
                    elevation={Elevation.TWO}
                    className="public-journal-card"
                    onClick={() => openJournalModal(journal)}
                  >
                    <h5>{journal.title || "Unbenannt"}</h5>
                    <p>
                      {journal.date
                        ? moment(journal.date).format("dddd, MMMM D, YYYY")
                        : "No Date Available"}
                    </p>
                    <div className="card-actions">
                      <Button
                        icon="thumbs-up"
                        text={`Like (${likes[journal.id] || 0})`}
                        onClick={(e) => {
                          e.stopPropagation();
                          handleLike(journal.id);
                        }}
                        active={userLiked[journal.id]}
                      />
                      <Button
                        icon="comment"
                        text="Comment"
                        onClick={(e) => {
                          e.stopPropagation();
                          openJournalModal(journal);
                        }}
                      />
                    </div>
                  </Card>
                );
              })
            ) : (
              <p>No public entries available</p>
            )}
          </div>
        </div>
      </div>
      <Dialog
        isOpen={isModalOpen}
        onClose={closeJournalModal}
        title={null}
        className={`modal-container ${darkMode ? "dark" : "bright"}`}
        style={{
          zIndex: 600,
          position: "fixed",
          top: "50%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          fontSize: "1.2em",
          backgroundColor: darkMode ? "#1c1c1e" : "#ffffff",
          color: darkMode ? "#e5e5ea" : "#000000",
          border: "1px solid #333",
          maxWidth: "90vw",
          maxHeight: "90vh",
          overflowY: "auto",
          boxShadow: "0 10px 30px rgba(0, 0, 0, 0.5)"
        }}
      >
        <div
          className="bp4-dialog-body"
          style={{ padding: "20px", lineHeight: "1.6" }}
        >
          <h2
            style={{
              color: darkMode ? "#ffffff" : "#000000",
              fontSize: "1.6em",
              fontWeight: "bold",
              textShadow: darkMode
                ? "1px 1px 3px rgba(0, 0, 0, 0.7)"
                : "1px 1px 3px rgba(200, 200, 200, 0.7)",
              marginBottom: "0.5em",
            }}
          >
            {selectedJournal?.title || "Öffentlich"}
          </h2>

          <p
            style={{
              fontSize: "1.1em",
              fontStyle: "italic",
              color: darkMode ? "#bbbbbb" : "#555555",
              marginBottom: "1.5em",
              padding: "5px 10px",
              borderRadius: "5px",
              border: darkMode ? "1px solid #444" : "1px solid #ddd",
              backgroundColor: darkMode ? "#2c2c2c" : "#f9f9f9",
            }}
          >
            <strong>Date:</strong> {selectedJournal?.formattedDate || "N/A"}
          </p>

          <div
            style={{
              backgroundColor: darkMode ? "#2a2a2a" : "#f0f0f0",
              padding: "10px",
              borderRadius: "8px",
              marginBottom: "1.5em",
              maxHeight: "200px",
              overflowY: "auto"
            }}
          >
            <p style={{ marginBottom: "0.5em", fontWeight: "bold" }}>
              <strong>Content:</strong>
            </p>
            {editorState && (
              <div className="editor-readonly">
                <Editor editorState={editorState} readOnly={true} />
              </div>
            )}
          </div>

          <div
            style={{
              backgroundColor: darkMode ? "#2a2a2a" : "#f0f0f0",
              padding: "10px",
              borderRadius: "8px",
            }}
          >
            <p style={{ marginBottom: "0.5em", fontWeight: "bold" }}>
              <strong>Comments:</strong>
            </p>
            <div className="comments-container">
              {comments.length > 0 ? (
                comments.map((comment: any, index: number) => (
                  <div key={index} style={{ marginBottom: "0.6em" }}>
                    <strong>{comment.username}:</strong>{" "}
                    {comment.deleted ? (
                      <span style={{ fontStyle: "italic" }}>
                        Comment was deleted by user
                      </span>
                    ) : (
                      <>
                        {comment.content}
                        {comment.edited && (
                          <span style={{ fontStyle: "italic", fontSize: "0.8em", color: "gray" }}>
                            {" "}
                            (edited)
                          </span>
                        )}
                      </>
                    )}
                    {comment.username === username && !comment.deleted && (
                      <span style={{ fontSize: "0.8em", color: "gray" }}>
                        <br />
                        {/* eslint-disable-next-line */}
                        <a href="#" onClick={() => handleEditComment(comment.id, comment.content)}>
                          Edit
                        </a>{" "}
                        |{" "}
                        {/* eslint-disable-next-line */}
                        <a href="#" onClick={() => handleDeleteComment(comment.id)}>
                          Delete
                        </a>
                      </span>
                    )}
                  </div>
                ))
              ) : (
                <p>No comments yet.</p>
              )}
            </div>
          </div>

          <textarea
            value={newComment}
            onChange={(e) => setNewComment(e.target.value)}
            placeholder="Add a comment..."
            className="comment-input"
            style={{
              width: "100%",
              padding: "10px",
              marginTop: "10px",
              borderRadius: "4px",
              border: "1px solid #ccc",
              backgroundColor: darkMode ? "#2c2c2e" : "#f9f9f9",
              color: darkMode ? "#e5e5ea" : "#000000",
            }}
          />
        </div>
        <div
          className="bp4-dialog-footer"
          style={{
            padding: "10px",
            display: "flex",
            justifyContent: "space-between",
          }}
        >
          <Button
            onClick={(e) => {
              e.stopPropagation();
              if (selectedJournal?.id) {
                handleLike(selectedJournal.id);
              }
            }}
            className={`like-button ${darkMode ? "dark-mode" : "light-mode"} ${
              userLiked[selectedJournal?.id] ? "liked" : ""
            }`}
          >
            Like ({likes[selectedJournal?.id] || 0})
          </Button>

          <div style={{ display: "flex", gap: "10px" }}>
            <Button
              onClick={handleCommentSubmit}
              style={{
                padding: "8px 16px",
                borderRadius: "4px",
                fontSize: "1.1em",
                backgroundColor: "#007bff",
                color: "#fff",
                transition: "background-color 0.3s ease",
              }}
              onMouseOver={(e) =>
                (e.currentTarget.style.backgroundColor = "#0056b3")
              }
              onMouseOut={(e) =>
                (e.currentTarget.style.backgroundColor = "#007bff")
              }
            >
              Submit
            </Button>
            <Button
              onClick={closeJournalModal}
              style={{
                padding: "8px 16px",
                borderRadius: "4px",
                fontSize: "1.1em",
                backgroundColor: "#555555",
                color: "#fff",
                transition: "background-color 0.3s ease",
              }}
              onMouseOver={(e) =>
                (e.currentTarget.style.backgroundColor = "#333333")
              }
              onMouseOut={(e) =>
                (e.currentTarget.style.backgroundColor = "#555555")
              }
            >
              Close
            </Button>
          </div>
        </div>
      </Dialog>
    </div>
  );
};

export default PublicJournals;
