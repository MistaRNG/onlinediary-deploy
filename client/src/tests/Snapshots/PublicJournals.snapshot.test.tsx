import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import PublicJournals from '../../features/journal/PublicJournals';
import journalReducer from '../../features/journal/journalSlice';
import { describe, it, expect } from '@jest/globals';

const mockStore = configureStore({
  reducer: {
    journals: journalReducer,
  },
  preloadedState: {
    journals: {
      data: {
        '2024-09-06': {
          id: 1,
          content: { blocks: [], entityMap: {} },
          title: 'Test Journal',
          text: 'Test Content',
          isPublic: true,
        },
      },
      gotData: true,
      saved: null,
      editCount: 0,
      prevDates: [],
    },
  },
});

describe('PublicJournals Component Snapshot Test', () => {
  it('renders correctly', () => {
    // ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <BrowserRouter>
          <PublicJournals />
        </BrowserRouter>
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
