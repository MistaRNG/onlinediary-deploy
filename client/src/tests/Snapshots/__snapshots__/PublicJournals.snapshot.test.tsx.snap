// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicJournals Component Snapshot Test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bright"
  >
    <div
      class="Logout-Button"
    >
      <button
        class="bp4-button"
        type="button"
      >
        <span
          aria-hidden="true"
          class="bp4-icon bp4-icon-people"
          icon="people"
        >
          <svg
            data-icon="people"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M13.69 13.98c-.05-.24-.14-.5-.25-.76-.36-.86-1.12-1.33-2.69-2-.14-.06-.59-.25-.6-.25-.21-.09-.36-.15-.5-.22.02-.1.02-.2.03-.31 0-.04.01-.08.01-.13-.07-.06-.13-.12-.19-.19.22-.32.4-.67.54-1.05.02-.06.02-.06.03-.1.29-.23.48-.57.59-.96.16-.33.25-.73.21-1.16-.03-.4-.16-.76-.37-1.03-.02-.53-.07-1.13-.15-1.54-.01-.06-.02-.12-.03-.19.23-.06.48-.09.72-.09.49 0 1.05.16 1.44.46.38.29.67.7.8 1.17.03.1.05.21.07.31.07.37.11.94.11 1.33v.05c.14.06.27.21.29.51.02.25-.07.45-.13.54-.05.21-.16.44-.38.48-.02.1-.05.2-.09.3 0 .01-.01.03-.01.03-.17.44-.43.83-.75 1.11v.14c.03.35-.09.59.83 1 .93.41 2.32.84 2.6 1.5.29.66.17 1.04.17 1.04h-2.3zm-1.17-.38c.37.86.22 1.36.22 1.36H.06s-.14-.5.22-1.36 2.13-1.43 3.31-1.96c1.17-.54 1.05-.86 1.09-1.3 0-.05.01-.11.01-.17-.41-.35-.75-.86-.97-1.45v-.01s-.01-.01-.01-.02c-.04-.12-.09-.26-.12-.39-.28-.05-.44-.36-.5-.64-.06-.12-.19-.39-.16-.71.04-.41.21-.6.39-.68v-.06c0-.51.05-1.26.14-1.74.02-.13.05-.27.09-.4.17-.6.54-1.13 1.02-1.51.5-.39 1.21-.6 1.84-.6s1.34.21 1.84.6c.48.38.85.91 1.02 1.52.04.13.07.27.09.4.09.48.14 1.22.14 1.73v.07c.18.08.34.27.37.67.03.32-.09.59-.16.71-.06.28-.21.58-.48.63-.03.13-.07.26-.12.39 0 .01-.01.04-.01.04-.22.58-.55 1.08-.95 1.45v.18c.04.45-.12.77 1.06 1.3 1.18.53 2.95 1.09 3.31 1.95z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp4-button-text"
        >
          Entries from others
        </span>
      </button>
      <button
        class="bp4-button"
        type="button"
      >
        <span
          class="bp4-button-text"
        >
          Logged in as @
        </span>
      </button>
      <button
        class="bp4-button"
        type="button"
      >
        <span
          aria-hidden="true"
          class="bp4-icon bp4-icon-log-out"
          icon="log-out"
        >
          <svg
            data-icon="log-out"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M7 14H2V2h5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v14c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zm8.71-6.71l-3-3a1.003 1.003 0 00-1.42 1.42L12.59 7H6c-.55 0-1 .45-1 1s.45 1 1 1h6.59l-1.29 1.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3-3c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="page-container"
    >
      <div
        class="header-container"
        style="position: fixed; filter: none; margin: 70px 0px 0px 1px; z-index: 1; display: block;"
      >
        <button
          class="bp4-button back-to-editor-button"
          style="margin-bottom: 30px; display: block;"
          type="button"
        >
          <span
            aria-hidden="true"
            class="bp4-icon bp4-icon-arrow-left"
            icon="arrow-left"
          >
            <svg
              data-icon="arrow-left"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp4-button-text"
          >
            Back to Editor
          </span>
        </button>
        <div
          class="bp4-input-group bp4-large search-input"
        >
          <input
            class="bp4-input"
            placeholder="Search..."
            style="padding-right: 0px;"
            type="text"
            value=""
          />
          <span
            class="bp4-input-action"
          />
        </div>
      </div>
      <h2
        style="text-align: center; width: 100%; margin-top: 20px; position: fixed; top: 30px; left: 50%; transform: translateX(-50%); z-index: 1;"
      >
        Public entries
      </h2>
      <div
        class="content-container"
        style="filter: none; margin-top: 100px; z-index: 0; position: relative;"
      >
        <div
          class="public-journals-container"
        >
          <div
            class="bp4-card bp4-interactive bp4-elevation-2 public-journal-card"
          >
            <h5>
              Test Journal
            </h5>
            <p>
              No Date Available
            </p>
            <div
              class="card-actions"
            >
              <button
                class="bp4-button"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="bp4-icon bp4-icon-thumbs-up"
                  icon="thumbs-up"
                >
                  <svg
                    data-icon="thumbs-up"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <path
                      d="M15.99 9.62c-.08-.58-.73-.96-1.43-1.05-.15-.02-.32-.04-.49-.05.06-.01.11-.01.16-.02.71-.08 1.23-.47 1.15-1.05-.08-.58-.73-.96-1.43-1.05-.34-.04-.72-.07-1.13-.09h-.01c-.24-.01-.49-.02-.75-.02-.48-.02-1.11-.04-1.59-.08-.36-.03-.41-.28-.37-.53.35-.9.54-1.91.5-2.91-.04-.85-.6-1.68-1.41-1.77-.67-.07-1.16.39-.99.98C8.76 3.91 6.13 6.2 4 7.49v6c1.66 1.03 4.35 1.56 7.48 1.5.19 0 .39-.01.62-.02h.09c.22-.01.46-.03.73-.05.71-.06 1.54-.51 1.5-1.09-.03-.51-.4-.79-.87-.93.27-.02.54-.04.79-.06.71-.06 1.24-.45 1.17-1.04-.06-.58-.7-.97-1.41-1.07-.02 0-.05-.01-.07-.01.29-.02.57-.03.81-.05.71-.03 1.23-.47 1.15-1.05zM2 7H0v7h2c.55 0 1-.45 1-1V8c0-.56-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp4-button-text"
                >
                  Like (0)
                </span>
              </button>
              <button
                class="bp4-button"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="bp4-icon bp4-icon-comment"
                  icon="comment"
                >
                  <svg
                    data-icon="comment"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <path
                      d="M14 1H1c-.55 0-1 .45-1 1v9c0 .55.45 1 1 1h2v3a1.003 1.003 0 001.71.71L8.41 12H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zM3.5 8C2.67 8 2 7.33 2 6.5S2.67 5 3.5 5 5 5.67 5 6.5 4.33 8 3.5 8zm4 0C6.67 8 6 7.33 6 6.5S6.67 5 7.5 5 9 5.67 9 6.5 8.33 8 7.5 8zm4 0c-.83 0-1.5-.67-1.5-1.5S10.67 5 11.5 5s1.5.67 1.5 1.5S12.33 8 11.5 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp4-button-text"
                >
                  Comment
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
